### PR TITLE
Use state

### PR DESCRIPTION
**Pull request by alejobh**
## Summary
* Added `useState` page example

## Screenshots

[Change!] Show the screenshots of the views you modified.

### CODE REVIEW

File: src/app/screens/Dashboard/screens/Home/index.tsx [hunk 1]
```diff
@@ -1,46 +1,23 @@
...
   return (
     <div className={styles.app}>
       <header className={styles.appHeader}>
         <img src={logo} className={styles.appLogo} alt="logo" />
-        <p className={styles.text}>You are logged in. {foo}</p>
-        <button type="button" className={styles.appLink} onClick={handleLogout}>
-          Logout
-        </button>
+        <p className={`${styles.text} m-bottom-6`}>React advanced hooks</p>
+        <NavLink to={paths.useState}>useEffect</NavLink>
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/Home/index.tsx]:
```suggestion
        <NavLink to={paths.useState}>useState</NavLink>
```
😅

File: src/app/screens/Dashboard/screens/UseState/components/SquareClassComponent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          <div className="column middle center square yellow m-right-4" onClick={this.handleSetSquare1}>
+            <p className="bold">Mouse position</p>
+            <p>Previous: {printCoords(square1.previous)}</p>
+            <p>Current: {printCoords(square1.current)}</p>
+            <p>Llamados: {square1Qty}</p>
+          </div>
+          <div className="column middle center square yellow" onClick={this.handleSetSquare2}>
+            <p className="bold">Mouse position</p>
+            <p>Position: {printCoords(square2)}</p>
+            <p>Llamados: {square2Qty}</p>
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseState/components/SquareClassComponent/index.tsx]: I'd write everything either in Spanish or English.

File: src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+
+import { SquareCoords } from '~utils/types';
+
+import { printCoords, initial } from '../../utils';
+
+function SquareFunctional() {
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx]:
```suggestion
function Square() {
```
We could keep de `ClassComponent` suffix, but I'd name functional ones as usual.

File: src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      return {
+        previous: prev.current,
+        current: { x: evt.nativeEvent.offsetX, y: evt.nativeEvent.offsetY }
+      };
+    });
+  };
+
+  const handleSetCoords2 = (evt: React.MouseEvent) => {
+    evt.persist();
+    // Multiple call
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx]:
```suggestion
    // Multiple calls
```

File: src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import { SquareCoords } from '~utils/types';
+
+import { printCoords, initial } from '../../utils';
+
+function SquareFunctional() {
+  const [square1, setSquare1] = useState<{
+    previous: SquareCoords;
+    current: SquareCoords;
+  }>(initial);
+  const [square2, setSquare2] = useState<SquareCoords>(initial.current);
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseState/components/SquareFunctional/index.tsx]: Is `<SquareCoords>` necessary? I think we can explain if it isn't in the demo

File: src/app/screens/Dashboard/screens/UseState/index.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import ScreenView from '~components/ScreenView';
+
+import SquareFunctional from './components/SquareFunctional';
+import SquareClassComponent from './components/SquareClassComponent';
+
+function UseState() {
+  return (
+    <ScreenView title="useState example">
+      <SquareFunctional />
+      <SquareClassComponent />
```
Comment (pabloferro) [inline src/app/screens/Dashboard/screens/UseState/index.tsx:12]: I'm not sure about showing class components 🤔 , we could leave them for now if you want.

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: src/app/screens/Dashboard/screens/UseState/index.tsx (hunk 6 above)
Comment (alejobh) [reply src/app/screens/Dashboard/screens/UseState/index.tsx:12]: We can keep it but not speak about it

**Review by pabloferro — APPROVED** [review]

**Pull request merged**